File: binary_search.js
const readline = require('readline-sync');
const process = require('process');
const perf = require('execution-time')();


function binary_search(arr,left,right,element) {

    if(left<=right) {

        var mid = (left+right)/2;

        if(arr[mid]===element) {
            return 1;
        }

        else if(element<arr[mid]) {
            binary_search(arr,left,mid-1,element);
        }
        
        return binary_search(arr,mid+1,right,element);
    }

    return 0;

}

var arr = [30,40,50,60,10,20];
var temp = 0;

for(var i=0;i<arr.length;i++) {
    for(var j=i+1;j<arr.length;j++) {
        if(arr[i]>arr[j]) {
            temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }
    }
}


console.log(arr);

var element = readline.questionInt("enter element : ");

console.log(element);

var result = binary_search(arr,0,arr.length-1,element);

if(result===1) console.log("element found.");   
else console.log("element not found.");


 

File: mayur/binary-search.cpp
#include <iostream>

using namespace std;

int main()
{

    int arr[5] = {1, 2, 3, 4, 5};

    int n = sizeof(arr) / sizeof(arr[0]);

    cout << "size of array is " << n << endl;

    int element;

    cout << "enter element to search in array : " << endl;

    cin >> element;

    int left = 0;
    int right = n - 1;
    int flag = 0;

    while (left <= right)
    {
        int middle = left + (right - left) / 2;
        if (arr[middle] == element)
        {
            cout << "element found at index " << middle << endl;
            flag = 1;
            break;
        }

        else
        {
            if (element < arr[middle])
            {
                right = middle - 1;
            }

            else
            {
                left = middle + 1;
            }
        }
    }

    if (flag == 0)
        cout << "element not found." << endl;
        
    return 0;
}
